### PR TITLE
Fix issue with #311

### PR DESCRIPTION
**Pull request by LynxAbraxas**
see https://github.com/civctp2/civctp2/pull/311#issuecomment-651999928 and following comments.

### CODE REVIEW

**Comment by LynxAbraxas** [comment]
@williamd67, @henshall828 what do you think? Please feel free to adjust.

**Comment by henshall828** [comment]
> 
> 
> @williamd67, @henshall828 what do you think? Please feel free to adjust.

The second part of the fix may not be quite right, as it could again lead to the problem with decrementing an iterator at the beginning of the list which leads to goal_iter containing an invalid value. A cleaner way to do this could be to have Repriotize_Goal return the iterator to use, then the code becomes
`goal_iter = Reprioritize_Goal(goal_iter)`
following the same pattern as the erase.

**Comment by LynxAbraxas** [comment]
@henshall828 please create a commit or PR that contains the changes.

File: ctp2_code/ai/strategy/scheduler/scheduler.cpp [hunk 1]
```diff
@@ -725,8 +725,11 @@ void Scheduler::Match_Resources(const bool move_armies)
 					// http://www.cplusplus.com/reference/stl/list/splice.html
 					//or use a decrement
 					// Sort the goal list, move iterator increment herein back
+					tmp_goal_iter = goal_iter;
+					--goal_iter;
```
Comment (henshall828) [inline ctp2_code/ai/strategy/scheduler/scheduler.cpp:729]: If goal_iter is the first item in the list of goals, this will cause an error.

**Comment by henshall828** [comment]
> 
> 
> @henshall828 please create a commit or PR that contains the changes.

Please can you take a look and test PR #337 

Many thanks @LynxAbraxas

**Comment by LynxAbraxas** [comment]
Closed in favor of #337.

**Pull request closed without merging**